src/zenml/integrations/huggingface/model_registries/huggingface_model_registry.py

```diff
@@ -0,0 +1,668 @@
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to get.
+
+        Returns:
+            The model version.
+        """
+        pr_num = version.split("/")[-1]
+        description, metadata, _ = self._get_model_card_content(
+            name=name, revision=version
+        )
+        return ModelVersion(
+            version=version,
+            model_source_uri=f"https://huggingface.co/{self._repo_id(name)}/discussions/{pr_num}",
+            model_format="hugginface",
+            registered_model=self.get_model(name),
+            description=description,
+            metadata=metadata,
+        )
+
+    def load_model_version(
+        self,
+        name: str,
+        version: str,
+        path: str,
+        **kwargs: Any,
+    ) -> None:
+        """Loads a model version from the model registry.
+
+        Args:
+            name: The name of the registered model.
+            version: The version of the model version to load.
+            path: The path to load model into.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            None.
+        """
+
+        files = self.api.list_repo_files(
+            repo_id=self._repo_id(name),
+            revision=version,
+            repo_type=self.REPO_TYPE,
+        )
+
+        download_folder_path = os.path.abspath(path)
+        if not os.path.exists(download_folder_path):
+            os.makedirs(download_folder_path)
+
+        for file in files:
+            self.api.hf_hub_download(
+                repo_id=self._repo_id(name),
+                filename=file,
+                local_dir=download_folder_path,
+                revision=version,
+            )
+
+    def get_model_uri_artifact_store(
+        self,
+        model_version: ModelVersion,
+    ) -> str:
+        """Gets the URI artifact store for a model version.
+
+        This method retrieves the URI of the artifact store for a specific model
+        version. Its purpose is to ensure that the URI is in the correct format
+        for the specific artifact store being used. This is essential for the
+        model serving component, which relies on the URI to serve the model
+        version. In some cases, the URI may be stored in a different format by
+        certain model registry integrations. This method allows us to obtain the
+        URI in the correct format, regardless of the integration being used.
+
+        Note: In some cases the URI artifact store may not be available to the
+        user, the method should save the target model in one of the other
+        artifact stores supported by ZenML and return the URI of that artifact
+        store.
+
+        Args:
+            model_version: The model version for which to get the URI artifact
+                store.
+
+        Returns:
+            The URI artifact store for the model version.
+        """
+        raise NotImplementedError
```

Comment:
long docstring for something that's not implemented?

Reply:
This is from Base model, I'm not sure how handy this method is at all, tbh